File: sdks/insuretech-go-sdk/pkg/models/endorsement.go
package models

import (
	"time"
)

// Endorsement represents a endorsement
type Endorsement struct {
	ApprovedAt time.Time `json:"approved_at,omitempty"`
	ApprovedBy string `json:"approved_by,omitempty"`
	AuditInfo interface{} `json:"audit_info"`
	Changes string `json:"changes"`
	EffectiveDate time.Time `json:"effective_date"`
	EndorsementNumber string `json:"endorsement_number"`
	Id string `json:"id"`
	PolicyId string `json:"policy_id"`
	PremiumAdjustment *Money `json:"premium_adjustment,omitempty"`
	PremiumRefundRequired bool `json:"premium_refund_required,omitempty"`
	Reason string `json:"reason"`
	RequestedBy string `json:"requested_by"`
	Status interface{} `json:"status"`
	Type *EndorsementType `json:"type"`
}

File: backend/insurance_engine/src/InsuranceEngine.Policy/Domain/Entities/Endorsement.cs
using System;
using InsuranceEngine.Policy.Domain.Enums;
using InsuranceEngine.SharedKernel.Interfaces;

namespace InsuranceEngine.Policy.Domain.Entities;

public class Endorsement
{
    public Guid Id { get; set; }

    public string EndorsementNumber { get; set; } = string.Empty;
    public Guid PolicyId { get; set; }
    public EndorsementType Type { get; set; }
    public string Reason { get; set; } = string.Empty;
    public string ChangesJson { get; set; } = "{}"; // Store before/after as JSON
    
    public decimal PremiumAdjustmentAmount { get; set; }
    public string PremiumAdjustmentCurrency { get; set; } = "BDT";
    public bool PremiumRefundRequired { get; set; }
    
    public EndorsementStatus Status { get; set; }
    public Guid RequestedBy { get; set; }
    public Guid? ApprovedBy { get; set; }
    
    public DateTime EffectiveDate { get; set; }
    public DateTime At { get; set; } // ApprovedAt
    public string AuditInfoJson { get; set; } = "{}";
    
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Navigation (Optional)
    // public virtual PolicyEntity Policy { get; set; } = null!;
}
